### PR TITLE
docs: improve concepts/security.rst

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add links
- add items
- remove a item that dose not make sense

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/concepts/security.rst [hunk 1]
```diff
@@ -38,8 +38,8 @@ OWASP recommendations
 CodeIgniter provisions
 ======================
 
-- :doc:`HTTP library <../incoming/incomingrequest>` provides for input field filtering & content metadata
-- Validation library
+- :doc:`HTTP library <../incoming/incomingrequest>` provides for :ref:`input field filtering <incomingrequest-filtering-input-data>` & content metadata
```
Comment (kenjis) [inline user_guide_src/source/concepts/security.rst:41]: What is *content metadata*?

File: user_guide_src/source/concepts/security.rst [hunk 2]
```diff
@@ -144,7 +147,7 @@ OWASP recommendations
 CodeIgniter provisions
 ======================
 
-- Session keys stored encrypted
+- :doc:`Session <../libraries/sessions>` keys stored encrypted
```
Comment (kenjis) [inline user_guide_src/source/concepts/security.rst:150]: What are *Session keys*?

Comment (kenjis) [reply user_guide_src/source/concepts/security.rst]: Removed it, and added some items.

**Comment by kenjis** [comment]
Review, please.

**Review by datamweb — APPROVED** [review]

**Pull request merged**